In [4]:
import pandas as pd
import numpy as np

In [5]:
orders = pd.read_csv(r"C:\Users\MAINAK\Desktop\Data-Science\projects\MarketLens\data\raw\olist_orders_dataset.csv")
orders.head()

,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00
3,949d5b44dbf5de918fe9c16f97b45f8a,f88197465ea7920adcdbec7375364d82,delivered,2017-11-18 19:28:06,2017-11-18 19:45:59,2017-11-22 13:39:59,2017-12-02 00:28:42,2017-12-15 00:00:00
4,ad21c59c0840e6cb83a9ceb5573f8159,8ab97904e6daea8866dbdbc4fb7aad2c,delivered,2018-02-13 21:18:39,2018-02-13 22:20:29,2018-02-14 19:46:34,2018-02-16 18:17:02,2018-02-26 00:00:00


In [6]:
print(orders.shape)
print(orders.columns)



(99441, 8)
Index(['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date'],
      dtype='str')


In [7]:
print(orders.dtypes)


order_id                         str
customer_id                      str
order_status                     str
order_purchase_timestamp         str
order_approved_at                str
order_delivered_carrier_date     str
order_delivered_customer_date    str
order_estimated_delivery_date    str
dtype: object


In [8]:
print(orders.isnull().sum())

order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64


In [9]:
orders["order_purchase_timestamp"] = pd.to_datetime(orders["order_purchase_timestamp"])
orders["order_approved_at"] = pd.to_datetime(orders["order_approved_at"])
orders["order_delivered_carrier_date"] = pd.to_datetime(orders["order_delivered_carrier_date"])
orders["order_delivered_customer_date"] = pd.to_datetime(orders["order_delivered_customer_date"])
orders["order_estimated_delivery_date"] = pd.to_datetime(orders["order_estimated_delivery_date"])


In [10]:
orders["order_status"].value_counts(normalize=True)


order_status
delivered      0.970203
shipped        0.011132
canceled       0.006285
unavailable    0.006124
invoiced       0.003158
processing     0.003027
created        0.000050
approved       0.000020
Name: proportion, dtype: float64

In [11]:
orders["order_purchase_timestamp"].min()

Timestamp('2016-09-04 21:15:19')

In [12]:
orders["order_purchase_timestamp"].max()

Timestamp('2018-10-17 17:30:18')

In [13]:
orders["delivery_days"] = orders["order_delivered_customer_date"]-orders["order_purchase_timestamp"]

In [14]:
orders["delivery_days"]

0        8 days 10:28:40
1       13 days 18:46:08
2        9 days 09:27:40
3       13 days 05:00:36
4        2 days 20:58:23
              ...       
99436    8 days 05:13:56
99437   22 days 04:38:58
99438   24 days 20:37:34
99439   17 days 02:04:27
99440    7 days 16:11:00
Name: delivery_days, Length: 99441, dtype: timedelta64[us]

In [15]:
#1 ans
len(orders[orders["order_status"] =="delivered"])


96478

In [16]:
orders.shape

(99441, 9)

In [17]:
#3 ans
delivered = len(orders[orders["order_status"] =="delivered"])
total_orders = len(orders)
print(total_orders)
print("Deliverd Orders in % :- ",(delivered / total_orders) *100)

99441
Deliverd Orders in % :-  97.02034372140264


In [18]:
orders["orders_months"] = orders["order_purchase_timestamp"].dt.to_period("M")
orders["orders_months"].head()

0    2017-10
1    2018-07
2    2018-08
3    2017-11
4    2018-02
Name: orders_months, dtype: period[M]

In [19]:
orders["orders_months"].sort_values()

4541     2016-09
10071    2016-09
4396     2016-09
30710    2016-09
10381    2016-10
          ...   
1801     2018-09
88500    2018-10
31891    2018-10
68373    2018-10
60938    2018-10
Name: orders_months, Length: 99441, dtype: period[M]

In [20]:
orders["delivery_days"].mean()

Timedelta('12 days 13:24:31.879068')

In [21]:
orders["delivery_days"].min()

Timedelta('0 days 12:48:07')

In [22]:
orders["delivery_days"].max()


Timedelta('209 days 15:05:12')

In [23]:

orders["Delayed_orders"] = (orders["order_delivered_customer_date"] > orders["order_estimated_delivery_date"])

In [24]:
orders["delay_days"]=(orders["order_delivered_customer_date"]-orders["order_estimated_delivery_date"]).dt.days
print(orders["delay_days"])

0        -8.0
1        -6.0
2       -18.0
3       -13.0
4       -10.0
         ... 
99436   -11.0
99437    -2.0
99438    -6.0
99439   -21.0
99440   -18.0
Name: delay_days, Length: 99441, dtype: float64


In [25]:
orders["delay_days"].mean()

np.float64(-11.876881296902857)

In [26]:
orders[orders["Delayed_orders"] == True]["delay_days"].mean()

np.float64(8.86725437587837)

# Business Insight 


In [27]:
# Total orders to delivered order is 97.020% with percentage
# Faser delivery all time is 0 days 12:48:07
# Sowest delivery all time to delivery orders  209 days 15:05:12
# Delayed orders average to delay day : - 8.87 days

## Check the revenue payments and sales for the trend according 


In [28]:
order_items = pd.read_csv(r"C:\Users\MAINAK\Desktop\Data-Science\projects\MarketLens\data\raw\olist_order_items_dataset.csv")
payments = pd.read_csv(r"C:\Users\MAINAK\Desktop\Data-Science\projects\MarketLens\data\raw\olist_order_payments_dataset.csv")

In [29]:
order_items.shape


(112650, 7)

In [30]:
order_items.columns


Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value'],
      dtype='str')

In [31]:
# Q1. Total product sales value
order_items["price"].sum()

np.float64(13591643.7)

In [32]:
# Q2. Total freight revenue/cost
order_items["freight_value"].sum()
#  all order to placed by company is 2251909

np.float64(2251909.54)

In [33]:

payments.shape

(103886, 5)

In [34]:
payments.columns


Index(['order_id', 'payment_sequential', 'payment_type',
       'payment_installments', 'payment_value'],
      dtype='str')

In [35]:
# Q3. Total payment value
payments["payment_value"].sum()
#  All total payments for to sale the product is :  16008872

np.float64(16008872.120000001)

In [36]:
#  Q4. Average product price
order_items["price"].mean()


np.float64(120.65373901464713)

In [37]:
# Q5. Most expensive product item
order_items["price"].max()
#  big amount for sale by company is 13664 

np.float64(6735.0)

In [38]:
# Q6. Cheapest product item
order_items["price"].min()
#  chepest price for the product the comany sale is 0 

np.float64(0.85)

In [39]:
# Q7. Payment method analysis
payments["payment_type"].value_counts()
# there are customer aretype of  payments_type during the purchasing product     credit_card    76795
#                                                                                boleto         19784
#                                                                                voucher         5775
#                                                                                debit_card      1529
#                                                                                not_defined        3

payment_type
credit_card    76795
boleto         19784
voucher         5775
debit_card      1529
not_defined        3
Name: count, dtype: int64

In [40]:
# Q8. Payment method percentage
payments["payment_type"].value_counts(normalize=True) * 100

payment_type
credit_card    73.922376
boleto         19.043952
voucher         5.558978
debit_card      1.471806
not_defined     0.002888
Name: proportion, dtype: float64

In [41]:
# Q9 — Installment analysis
payments["payment_installments"].min()


np.int64(0)

In [42]:
payments["payment_installments"].max()


np.int64(24)

In [43]:

payments["payment_installments"].mean()


np.float64(2.853348863176944)

In [44]:
products = pd.read_csv(r"C:\Users\MAINAK\Desktop\Data-Science\projects\MarketLens\data\raw\olist_products_dataset.csv")

In [45]:
products.shape

(32951, 9)

In [46]:
products.columns

Index(['product_id', 'product_category_name', 'product_name_lenght',
       'product_description_lenght', 'product_photos_qty', 'product_weight_g',
       'product_length_cm', 'product_height_cm', 'product_width_cm'],
      dtype='str')

In [47]:
# 1️⃣ Product Categories
products["product_category_name"].nunique
#  73
# business insight 
#  there are check a how many products name in table is 73 

<bound method IndexOpsMixin.nunique of 0                               perfumaria
1                                    artes
2                            esporte_lazer
3                                    bebes
4                    utilidades_domesticas
                       ...                
32946                     moveis_decoracao
32947    construcao_ferramentas_iluminacao
32948                      cama_mesa_banho
32949               informatica_acessorios
32950                      cama_mesa_banho
Name: product_category_name, Length: 32951, dtype: str>

In [48]:
#  2️⃣ Missing Categories
products["product_category_name"].isnull().sum()
#  610 
# business insight
#  that are 610 rows are affected to not complete data fill that

np.int64(610)

In [49]:
# 3️⃣ Category Distribution
products["product_category_name"].value_counts().head(10)

# product_category_name
# cama_mesa_banho           3029
# esporte_lazer             2867
# moveis_decoracao          2657
# beleza_saude              2444
# utilidades_domesticas     2335
# automotivo                1900
# informatica_acessorios    1639
# brinquedos                1411
# relogios_presentes        1329
# telefonia                 1134
# Name: count, dtype: int64

#  business insight 
#  product categorey wise top 10 most selling  product name 

product_category_name
cama_mesa_banho           3029
esporte_lazer             2867
moveis_decoracao          2657
beleza_saude              2444
utilidades_domesticas     2335
automotivo                1900
informatica_acessorios    1639
brinquedos                1411
relogios_presentes        1329
telefonia                 1134
Name: count, dtype: int64

In [50]:
# 4️⃣ Product Photos
products["product_photos_qty"].mean()
# np.float64(2.1889861166939797)

#  business insight 
#  there are most of the product average  pic in the table is 2 

np.float64(2.1889861166939797)

In [51]:
# 5️⃣ Most Photos
products["product_photos_qty"].max()
# np.float64(20.0)

# business insight 
#  maximum images for one product in the table appoximently 20 

np.float64(20.0)

In [52]:
#  6️⃣ Product Weight
products["product_weight_g"].mean()
# np.float64(2276.4724877841513)

#  business insight 
# there are average weight of product 2.3kg

np.float64(2276.4724877841513)

In [53]:
# 7️⃣ Heaviest Product
products["product_weight_g"].max()
#  np.float64(40425.0)

#  business insight 
#  there are maximum weight for product is 4kg

np.float64(40425.0)

In [54]:
#  8️⃣ Product Dimensions
products[["product_height_cm",
         "product_length_cm",
         "product_width_cm"]].mean()

# product_height_cm    16.937661
# product_length_cm    30.815078
# product_width_cm     23.196728
# dtype: float64

#  business insight 
#  there are check the average product size to  calulate in cm

product_height_cm    16.937661
product_length_cm    30.815078
product_width_cm     23.196728
dtype: float64

In [55]:
# 9️⃣ Missing Data Check
products.isnull().sum()

# product_id                      0
# product_category_name         610
# product_name_lenght           610
# product_description_lenght    610
# product_photos_qty            610
# product_weight_g                2
# product_length_cm               2
# product_height_cm               2
# product_width_cm                2
# dtype: int64

#  business insight 
#  column wise to check a all empty value in table 

product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

## Product + Sales Analysis


In [56]:
order_items.columns

Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value'],
      dtype='str')

In [57]:
# Q1. Total Products Sold
order_items.groupby("product_id").size().reset_index(name = "total_itms_sold")

,product_id,total_itms_sold
0,00066f42aeeb9f3007548bb9d3f33c38,1
1,00088930e925c41fd95ebfe695fd2655,1
2,0009406fd7479715e4bef61dd91f2462,1
3,000b8f95fcb9e0096488278317764d19,2
4,000d9be29b5207b54e86aa1b1ac54872,1
...,...,...
32946,fff6177642830a9a94a0f2cba5e476d1,2
32947,fff81cc3158d2725c0655ab9ba0f712c,1
32948,fff9553ac224cec9d15d49f5a263411f,1
32949,fffdb2d0ec8d6a61f0a0a0db3f25b441,5


In [58]:
#  Q2. Top 10 Most-Sold Product
product_counts = order_items.groupby("product_id")["order_item_id"].count()

sorted_counts = product_counts.sort_values(ascending = False).head(10)
sorted_counts

product_id
aca2eb7d00ea1a7b8ebd4e68314663af    527
99a4788cb24856965c36a24e339b6058    488
422879e10f46682990de24d770e7f83d    484
389d119b48cf3043d311335e499d9c6b    392
368c6c730842d78016ad823897a372db    388
53759a2ecddad2bb87a079a1f1519f73    373
d1c427060a0f73f6b889a5c7c61f2ac4    343
53b36df67ebb7c41585e8d54d6772e08    323
154e7e31ebfa092203795c972e5804a6    281
3dd2a17168ec895c781a9191c1e95ad7    274
Name: order_item_id, dtype: int64

In [59]:
# Q3. Product-wise Total Revenue
sales_revenue = order_items.groupby("product_id")["price"].sum()
print(sales_revenue)

product_id
00066f42aeeb9f3007548bb9d3f33c38    101.65
00088930e925c41fd95ebfe695fd2655    129.90
0009406fd7479715e4bef61dd91f2462    229.00
000b8f95fcb9e0096488278317764d19    117.80
000d9be29b5207b54e86aa1b1ac54872    199.00
                                     ...  
fff6177642830a9a94a0f2cba5e476d1    229.98
fff81cc3158d2725c0655ab9ba0f712c     90.00
fff9553ac224cec9d15d49f5a263411f     32.00
fffdb2d0ec8d6a61f0a0a0db3f25b441    169.95
fffe9eeff12fcbd74a2f2b007dde0c58    249.99
Name: price, Length: 32951, dtype: float64


In [60]:
#  Q4. Top 10 Products by Revenue
sales_revenue.sort_values(ascending= False).head(10)

product_id
bb50f2e236e5eea0100680137654686c    63885.00
6cdd53843498f92890544667809f1595    54730.20
d6160fb7873f184099d9bc95e30376af    48899.34
d1c427060a0f73f6b889a5c7c61f2ac4    47214.51
99a4788cb24856965c36a24e339b6058    43025.56
3dd2a17168ec895c781a9191c1e95ad7    41082.60
25c38557cf793876c5abdd5931f922db    38907.32
5f504b3a1c75b73d6151be81eb05bdc9    37733.90
53b36df67ebb7c41585e8d54d6772e08    37683.42
aca2eb7d00ea1a7b8ebd4e68314663af    37608.90
Name: price, dtype: float64

In [61]:
# Q5. Category-wise Items Sold
product_sales = products.merge(
    order_items,
    on = "product_id"
)
product_sales.groupby("product_category_name")["order_item_id"].count()


product_category_name
agro_industria_e_comercio     212
alimentos                     510
alimentos_bebidas             278
artes                         209
artes_e_artesanato             24
                             ... 
sinalizacao_e_seguranca       199
tablets_impressao_imagem       83
telefonia                    4545
telefonia_fixa                264
utilidades_domesticas        6964
Name: order_item_id, Length: 73, dtype: int64

In [62]:
# Q6. Category-wise Revenue

Revenue_Categorey = product_sales.groupby("product_category_name")["price"].sum()

In [63]:
# Q7. Top 10 Categories by Revenue
Revenue_Categorey.sort_values(ascending=False).head(10)

product_category_name
beleza_saude              1258681.34
relogios_presentes        1205005.68
cama_mesa_banho           1036988.68
esporte_lazer              988048.97
informatica_acessorios     911954.32
moveis_decoracao           729762.49
cool_stuff                 635290.85
utilidades_domesticas      632248.66
automotivo                 592720.11
ferramentas_jardim         485256.46
Name: price, dtype: float64

In [64]:
# Q8. Top 10 Products by Quantity Sold
product_sales.groupby("product_id")["order_item_id"].count().sort_values(ascending=False).head(10)

product_id
aca2eb7d00ea1a7b8ebd4e68314663af    527
99a4788cb24856965c36a24e339b6058    488
422879e10f46682990de24d770e7f83d    484
389d119b48cf3043d311335e499d9c6b    392
368c6c730842d78016ad823897a372db    388
53759a2ecddad2bb87a079a1f1519f73    373
d1c427060a0f73f6b889a5c7c61f2ac4    343
53b36df67ebb7c41585e8d54d6772e08    323
154e7e31ebfa092203795c972e5804a6    281
3dd2a17168ec895c781a9191c1e95ad7    274
Name: order_item_id, dtype: int64

In [65]:
# # Q9. Find the top 10 products by total revenue.
product_sales.groupby("product_id")["price"].sum().sort_values(ascending = False).head(10)

product_id
bb50f2e236e5eea0100680137654686c    63885.00
6cdd53843498f92890544667809f1595    54730.20
d6160fb7873f184099d9bc95e30376af    48899.34
d1c427060a0f73f6b889a5c7c61f2ac4    47214.51
99a4788cb24856965c36a24e339b6058    43025.56
3dd2a17168ec895c781a9191c1e95ad7    41082.60
25c38557cf793876c5abdd5931f922db    38907.32
5f504b3a1c75b73d6151be81eb05bdc9    37733.90
53b36df67ebb7c41585e8d54d6772e08    37683.42
aca2eb7d00ea1a7b8ebd4e68314663af    37608.90
Name: price, dtype: float64

In [66]:
# Q10. Find the average price of products in each category.
product_sales.groupby("product_category_name")["price"].mean()

product_category_name
agro_industria_e_comercio    342.124858
alimentos                     57.634137
alimentos_bebidas             54.602446
artes                        115.802105
artes_e_artesanato            75.583750
                                ...    
sinalizacao_e_seguranca      108.086583
tablets_impressao_imagem      90.703735
telefonia                     71.213978
telefonia_fixa               225.693182
utilidades_domesticas         90.788148
Name: price, Length: 73, dtype: float64

In [67]:
# Q11. Find the top 10 sellers by total revenue.
product_sales.groupby("seller_id")["price"].sum().sort_values(ascending= False).head(10)

seller_id
4869f7a5dfa277a7dca6462dcf3b52b2    229472.63
53243585a1d6dc2643021fd1853d8905    222776.05
4a3ca9315b744ce9f8e9374361493884    200472.92
fa1c13f2614d7b5c4749cbc52fecda94    194042.03
7c67e1448b00f6e969d365cea6b010ab    187923.89
7e93a43ef30c4f03f38b393420bc753a    176431.87
da8622b14eb17ae2831f4ac5b9dab84a    160236.57
7a67c85e85bb2ce8582c35f2203ad736    141745.53
1025f0e2d44d7041d6cf58b6550e0bfa    138968.55
955fee9216a65b617aa5c0531780ce60    135171.70
Name: price, dtype: float64

In [68]:
# Q12. Find the top 10 sellers by number of items sold.
product_sales.columns

Index(['product_id', 'product_category_name', 'product_name_lenght',
       'product_description_lenght', 'product_photos_qty', 'product_weight_g',
       'product_length_cm', 'product_height_cm', 'product_width_cm',
       'order_id', 'order_item_id', 'seller_id', 'shipping_limit_date',
       'price', 'freight_value'],
      dtype='str')

In [69]:
product_sales.groupby("seller_id")["order_item_id"].count().sort_values(ascending = False).head(10)

seller_id
6560211a19b47992c3666cc44a7e94c0    2033
4a3ca9315b744ce9f8e9374361493884    1987
1f50f920176fa81dab994f9023523100    1931
cc419e0650a3c5ba77189a1882b7556a    1775
da8622b14eb17ae2831f4ac5b9dab84a    1551
955fee9216a65b617aa5c0531780ce60    1499
1025f0e2d44d7041d6cf58b6550e0bfa    1428
7c67e1448b00f6e969d365cea6b010ab    1364
ea8482cd71df3c1969d7b9473ff13abc    1203
7a67c85e85bb2ce8582c35f2203ad736    1171
Name: order_item_id, dtype: int64

In [70]:
# Q13. Find the average product price for each seller.
product_sales.groupby("seller_id")["price"].mean()

seller_id
0015a82c2db000af6aaaf3ae2ecb0532    895.000000
001cca7ae9ae17fb1caed9dfb1094831    104.937364
001e6ad469a905060d959994f1b41e4f    250.000000
002100f778ceb8431b7a1020ff7ab48f     22.445455
003554e2dce176b5555353e4f3555ac8    120.000000
                                       ...    
ffcfefa19b08742c5d315f2791395ee5     69.900000
ffdd9f82b9a447f6f8d4b91554cc7dd3    105.060000
ffeee66ac5d5a62fe688b9d26f83f534    131.418571
fffd5413c0700ac820c7069d66d98c89    148.562295
ffff564a4f9085cd26170f4732393726     71.315000
Name: price, Length: 3095, dtype: float64

In [71]:
#  radhe radhe 


In [74]:
#  Q2 — Unsold Products
products.columns

Index(['product_id', 'product_category_name', 'product_name_lenght',
       'product_description_lenght', 'product_photos_qty', 'product_weight_g',
       'product_length_cm', 'product_height_cm', 'product_width_cm'],
      dtype='str')

In [75]:
order_items.columns

Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value'],
      dtype='str')

In [ ]:
unsold_check = products.merge(
    order_items,
    on="product_id",
    how="left"
)
unsold_check["order_item_id"].isnull().sum()

(112650, 15)

In [82]:

unsold_check["order_item_id"].isnull().sum()

np.int64(0)

In [ ]:
unsold_products = unsold_check[
    unsold_check["order_item_id"].isnull()
]

In [80]:
print(unsold_products)

Empty DataFrame
Columns: [product_id, product_category_name, product_name_lenght, product_description_lenght, product_photos_qty, product_weight_g, product_length_cm, product_height_cm, product_width_cm, order_id, order_item_id, seller_id, shipping_limit_date, price, freight_value]
Index: []


In [93]:
# Q1: Total Sellers
order_items["seller_id"].nunique()

3095

In [96]:
order_items.columns

Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value'],
      dtype='str')

In [98]:
# Q2 — Seller-wise Orders.
order_sales = order_items.groupby("seller_id")["order_id"].count()

In [ ]:
# Q3 — Top 10 sellers by number of items sold.
order_sales.sort_values(ascending = False).head(10)

seller_id
6560211a19b47992c3666cc44a7e94c0    2033
4a3ca9315b744ce9f8e9374361493884    1987
1f50f920176fa81dab994f9023523100    1931
cc419e0650a3c5ba77189a1882b7556a    1775
da8622b14eb17ae2831f4ac5b9dab84a    1551
955fee9216a65b617aa5c0531780ce60    1499
1025f0e2d44d7041d6cf58b6550e0bfa    1428
7c67e1448b00f6e969d365cea6b010ab    1364
ea8482cd71df3c1969d7b9473ff13abc    1203
7a67c85e85bb2ce8582c35f2203ad736    1171
Name: order_id, dtype: int64

In [ ]:
# Q4 — Seller-wise Revenue
seller_revenue = order_items.groupby("seller_id")["price"].sum()

In [103]:
#  Q5 — Top 10 sellers by revenue.
seller_revenue.sort_values(ascending = False).head(10)

seller_id
4869f7a5dfa277a7dca6462dcf3b52b2    229472.63
53243585a1d6dc2643021fd1853d8905    222776.05
4a3ca9315b744ce9f8e9374361493884    200472.92
fa1c13f2614d7b5c4749cbc52fecda94    194042.03
7c67e1448b00f6e969d365cea6b010ab    187923.89
7e93a43ef30c4f03f38b393420bc753a    176431.87
da8622b14eb17ae2831f4ac5b9dab84a    160236.57
7a67c85e85bb2ce8582c35f2203ad736    141745.53
1025f0e2d44d7041d6cf58b6550e0bfa    138968.55
955fee9216a65b617aa5c0531780ce60    135171.70
Name: price, dtype: float64

In [105]:
order_items.columns

Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value'],
      dtype='str')

In [107]:
# Q6 — Average Selling Price per Seller
avg_selling_price = order_items.groupby("seller_id")["price"].mean()

In [109]:
#  Q7 — Seller-wise Freight Revenue
seller_frieght = order_items.groupby("seller_id")["freight_value"].sum()

In [ ]:
# Q8 — Seller Order Contribution %

seller_contribution = (order_sales / (order_sales.sum())) *100


In [118]:
#  Q9 — Sellers with Very Low Sales
order_sales.describe()
low_sales_sellers = order_sales[
    order_sales <= 2
]
low_sales_sellers.shape

(837,)

### ### Seller Analysis — Business Observations

1. **Total Sellers:** The dataset contains **3,095 unique sellers**, indicating a large and diverse seller base.

2. **Seller-wise Orders:** Seller order volumes vary significantly, with some sellers handling many more order-item records than others.

3. **Top 10 Sellers by Items Sold:** A small group of sellers contributes a large number of sold items, indicating that sales volume is not evenly distributed across all sellers.

4. **Seller-wise Revenue:** Revenue differs considerably between sellers because both the number of items sold and their selling prices vary.

5. **Top 10 Sellers by Revenue:** The highest-revenue sellers represent the sellers generating the greatest total sales value and are important contributors to overall marketplace revenue.

6. **Average Selling Price per Seller:** Average selling prices differ across sellers, showing that sellers operate across different product price ranges.

7. **Seller-wise Freight Revenue:** Freight value also varies between sellers, depending on the number of items sold and the freight charges associated with their orders.

8. **Seller Order Contribution:** Seller contribution percentages show how much each seller contributes to the overall order-item volume. This helps identify whether marketplace activity is concentrated among a smaller group of sellers.

9. **Very Low Sales Sellers:** The sales distribution shows a **median of only 8 order-item records per seller**, while the minimum is 1. This indicates that many sellers have relatively low sales volume.

10. **Overall Business Observation:** The seller base is large, but sales are **unevenly distributed**. A smaller group of sellers generates substantially higher sales volumes, while many sellers operate with relatively low order activity. This suggests that seller performance varies considerably across the marketplace.
